addons/binding/org.openhab.binding.tplinksmarthome/src/main/java/org/openhab/binding/tplinksmarthome/internal/TPLinkSmartHomeDiscoveryService.java

```diff
@@ -0,0 +1,192 @@
+                    }
+                }
+            } catch (SocketTimeoutException e) {
+                logger.debug("Discovering poller timeout...");
+            } catch (IOException e) {
+                logger.debug("Error during discovery: {}", e.getMessage());
+            } finally {
+                closeDiscoverSocket();
+                removeOlderResults(getTimestampOfLastScan());
+            }
+        }
+    }
+
+    /**
+     * Opens a {@link DatagramSocket} and sends a packet for discovery of TP-Link Smart Home devices.
+     *
+     * @return Returns the new socket
+     * @throws IOException exception in case sending the packet failed
+     */
+    protected DatagramSocket sendDiscoveryPacket() throws IOException {
+        DatagramSocket ds = new DatagramSocket(null);
+        ds.setBroadcast(true);
+        ds.setSoTimeout(UDP_PACKET_TIMEOUT);
+        ds.send(discoverPacket);
+        if (logger.isTraceEnabled()) {
+            logger.trace("Discovery package sent: {}", new String(discoverPacket.getData(), StandardCharsets.UTF_8));
+        }
+        return ds;
+    }
+
+    /**
+     * Closes the discovery socket and cleans the value. No need for synchronization as this method is called from a
+     * synchronized context.
+     */
+    private void closeDiscoverSocket() {
+        if (discoverSocket != null) {
+            discoverSocket.close();
+            discoverSocket = null;
+        }
+    }
+
+    /**
+     * Detected a device (thing) and get process the data from the device and report it discovered.
+     *
+     * @param packet containing data of detected device
+     * @throws IOException in case decrypting of the data failed
+     */
+    private void detectThing(DatagramPacket packet) throws IOException {
+        String ipAddress = packet.getAddress().getHostAddress();
+        String rawData = CryptUtil.decrypt(packet.getData(), packet.getLength());
+        Sysinfo sysinfo = commands.getSysinfoReponse(rawData);
+
+        logger.trace("Detected TP-Link Smart Home device: {}", rawData);
+        String deviceId = sysinfo.getDeviceId();
+        logger.debug("TP-Link Smart Home device '{}' with id {} found on {} ", sysinfo.getAlias(), deviceId, ipAddress);
+        Optional<ThingTypeUID> thingTypeUID = getThingTypeUID(sysinfo.getModel());
+
+        if (thingTypeUID.isPresent()) {
+            ThingUID thingUID = new ThingUID(thingTypeUID.get(),
+                    deviceId.substring(deviceId.length() - 6, deviceId.length()));
+            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withLabel(sysinfo.getAlias())
+                    .withProperties(PropertiesCollector.collectProperties(thingTypeUID.get(), ipAddress, sysinfo))
+                    .build();
+            thingDiscovered(discoveryResult);
+        } else {
+            logger.debug("Detected, but ignoring unsupported TP-Link Smart Home device model '{}'", sysinfo.getModel());
```

Comment:
Maybe this can be an info command, @kaikreuzer wdyt?

Reply:
ok for me

Reply:
I'll leave it with debug. Because autodiscovery is supported in this binding it would trigger an info log each time the discovery process runs an finds a unsupported device. This would be to much logging imho